README.md
# terraProjects

Before working on any of these projects, the following actions are required:
1) Create an S3 bucket named "myterraprojects"
2) Create a new key pair named "simpleVPC.pem" and download it to /c/dev/awsKeyPairDir
3) From command line, perform "aws configure"
4) From command line, in workspace, "terraform init -reconfigure" 

EX: 
Create bucket - aws s3 mb s3://myterraprojects --region us-east-1

Generate Key Pair - aws ec2 create-key-pair --key-name simpleVPC --query 'KeyMaterial' --output text > ../../awsKeyPairDir/simpleVPC.pem


history
    1  sudo yum update -y 
    2  sudo yum install epel-release -y 
    3  sudo yum install ansible -y 
    4  sudo vi /etc/ansible/hosts 
    5  id addr 
    6  ip addr 
    7  sudo vi /etc/ansible/hosts 
    8  ansible all -m ping 
    9  sudo vi /etc/ansible/hosts
   10  ansible all -m ping 
   11  ssh-keygen 
   12  ansible all -m ping 
   13  v ~/.ssh/known_hosts 
   14  vi  ~/.ssh/known_hosts 
   15  ansible all -m ping 
   16  vi  ~/.ssh/known_hosts 
   17  sudo vi /etc/ansible/hosts
   18  ansible all -m ping 
   19  ansible localhost -m ping 
   20  clear
   21  sudo yum update -y 
   22  sudo yum install wget unzip -y 
   23  sudo vi /etc/ansible/hosts
   24  ansible -m ping 
   25  ansible all -m ping 
   26  sudo yum install wget unzip -y 
   27  wget https://releases.hashicorp.com/terraform/1.3.4/terraform_1.3.4_linux_amd64.zip 
   28  unzip terraform_1.3.4_linux_amd64.zip 
   29  ls -lrt 
   30  sudo mv terraform /usr/local/bin
   31  terraform version 
   32  which git 
   33  pwd
   34  git clone https://github.com/mayeeyelekeri/terraProjects.git 
   35  pwd
   36  ls -lrt 
   37  rm -rf terraform_1.3.4_linux_amd64.zip 
   38  cd terraProjects/
   39  ls -lrt 
   40  cd applicationLBPrivatePublic/
   41  pwd
   42  cd ..
   43  git clone https://github.com/xchangebit/ansible-aws-cli 
   44  cd ansible-aws-cli/
   45  ls -lrt 
   46  cat playbook.yml 
   47  pwd
   48  cd ..
   49  ls -lrt 
   50  vi .bashrc 
   51  source ~/.bashrc 
   52  ls -lrt 
   53  vi .bashrc 
   54  source ~/.bashrc 
   55  cd ansible-aws-cli/
   56  ls -lrt 
   57  ap playbook.yml --check 
   58  su
   59  sudo - 
   60  sudo su 
   61  vi playbook.yml 
   62  ap playbook.yml --check 
   63  vi playbook.yml 
   64  ap playbook.yml --check 
   65  vi playbook.yml 
   66  ap playbook.yml --check 
   67  ap playbook.yml
   68  sudo su 
   69  ls -lrt 
   70  ap playbook.yml
   71  sudo yum install python 
   72  ls -lrt 
   73  cp playbook.yml myplaybook.yml
   74  vi myplaybook.yml 
   75  ap myplaybook.yml 
   76  vi myplaybook.yml 
   77  ap myplaybook.yml 
   78  vi myplaybook.yml 
   79  ap myplaybook.yml 
   80  python -version 
   81  python
   82  which python
   83  which python39
   84  pwd
   85  ls -lrt 
   86  ls -lrt /usr/bin/python 
   87  ls -lrt /usr/local/bin/python 
   88  ls -lrt /usr/local/bin/python*
   89  which unzip 
   90  v myplaybook.yml 
   91  vi  myplaybook.yml 
   92  ap myplaybook.yml 
   93  vi  myplaybook.yml 
   94  ap myplaybook.yml 
   95  aws 
   96  which awscli 
   97  ls -lrt /usr/bin 
   98  ls -lrt /usr/bin/aws 
   99  wget https://s3.amazonaws.com/aws-cli/awscli-bundle.zip dest=/tmp/awscli-bundle.zip 
  100  ls -lrt 
  101  unzip awscli-bundle.zip 
  102  cd awscli-bundle/
  103  ls -lrt 
  104  which python3 
  105  sudo 
  106  sudo ln -l /usr/bin/python /usr/bin/python3 
  107  sudo ln /usr/bin/python /usr/bin/python3 
  108  sudo ln /usr/bin/python3 /usr/bin/python 
  109  which pythong 
  110  which python
  111  ls -lrt 
  112  python install -i /usr/local/aws -b /usr/bin/aws 
  113  which aws 
  114  sudo yum install python
  115  sudo yum install python39 
  116  which python 
  117  python -v 
  118  pwd
  119  ls -lrt 
  120  python install -i /usr/local/aws -b /usr/bin/aws 
  121  rm -rf /usr/bin/python 
  122  sudo rm -rf /usr/bin/python 
  123  ls -lrt /usr/bin/python*
  124  sudo ln /usr/bin/python3.9 /usr/bin/python
  125  python version 
  126  python ver
  127  python -version 
  128  python -ver
  129  python -v
  130  sudo python install -i /usr/local/aws -b /usr/bin/aws 
  131  aws -version 
  132  aws version
  133  aws configure 
  134  pwd
  135  cd ..
  136  pwd
  137  cd ..
  138  ls -lrt 
  139  cd terraProjects/
  140  ls
  141  cd applicationLBPrivatePublic/
  142  ls -lrt 
  143  cat awsStorage.tf 
  144  cat init.sh 
  145  aws s3 mb s3://myterraprojects1 --region us-east-1
  146  t -version 
  147  pwd
  148  ls -lrt 
  149  tinit 
  150  tdev
  151  ls dev 
  152  tplan -var-file=dev/terraform.tfvars 
  153  tapply -var-file=dev/terraform.tfvars 
  154  history 
  155  exit
  156  pwd
  157  ls -lrt 
  158  cd terraProjects/applicationLBPrivatePublic/
  159  ls -lrt 
  160  ssh ec2-user@44.193.214.16
  161  pwd
  162  ls -lrt 
  163  ls -lrt /
  164  ls /srv 
  165  ls -lrt /src 
  166  exit
  167  ls -lrt /src 
  168  pwd
  169  ls -lrt 
  170  cd /src
  171  ls -lrt 
  172  cd terraProjects/
  173  ls -lrt 
  174  cd
  175  pwd
  176  ls -lrt 
  177  cd terraProjects/
  178  h
  179  history 
  180  tdestroy -var-file=dev/terraform.tfvars 
  181  cd applicationLBPrivatePublic/
  182  tdestroy -var-file=dev/terraform.tfvars 
  183  pwd
  184  cd /sc
  185  ls -lrt 
  186  pwd
  187  cd /src
  188  ls
  189  cd terraProjects/
  190  cd applicationLBPrivatePublic/
  191  ls -lrt 
  192  tinit  
  193  tdev
  194  pwd
  195  ls dev
  196  tapply -var-file=dev/terraform.tfvars
  197  pwd
  198  ls -lrt 
  199  pwd
  200  chmod -R 777 . 
  201  tapply -var-file=dev/terraform.tfvars
  202  chmod -R 755 . 
  203  tapply -var-file=dev/terraform.tfvars
  204  pwd
  205  ls -lrt 
  206  chmod o-w . 
  207  tapply -var-file=dev/terraform.tfvars
  208  pwd
  209  ls -lrt 
  210  ls -ld . 
  211  pwd
  212  ls -lrt 
  213  mkdir ansible
  214  ls -lrt 
  215  chmod 755 ansile 
  216  chmod 755 ansible
  217  ls -lrt 
  218  sudo chmod o-w ansible 
  219  ls -lrt 
  220  tdestroy  -var-file=dev/terraform.tfvars
  221  tapply  -var-file=dev/terraform.tfvars
  222  ansible-playbook --extra-vars 'ANSIBLE_CONFIG=ansible.cfg passed_in_hosts=jenkinsserver' ansible_templates/install_jenkins.yaml 
  223  ANSIBLE_CONFIG=ansible.cfg
  224  ansible-playbook --extra-vars 'ANSIBLE_CONFIG=ansible.cfg passed_in_hosts=jenkinsserver' ansible_templates/install_jenkins.yaml 
  225  echo $ANSIBLE_CONFIG
  226  ls
  227  exit
  228  ssh ec2-user@34.200.228.91
  229  ssh ec2-user@3.221.163.66
  230  pwd
  231  ls -lrt 
  232  cd terraProjects/

springdemo/setup.sh

--- Create an S3 bucket for strong terraform state 
aws s3 mb s3://terraprojects --region us-east-1

--- In database directory, execute script to create mysql creds in secrets manager 
. ~/INFO/secrets/aws_secrets.sh 

--- Make sure the web application source directory exists at this location where the terraform is running from 
/home/vagrant/SpringDataTest and /home/vagrant/Client exists 

--- Make sure "ansible" is installed 

--- Key-pair is from the local machine
  *** should be able to connect to EC2 instance using the command: 
   ssh -i ~/.ssh/id_rsa  ec2-user@<instance> 

--- This project is going to create 2 ec2s - one for server and other for client. 

--- This project depends on database is already created with data  

--- In Client module, update Servers IP address at application.properties 

--- To execute 
terraform init -reconfigure 
terraform apply -var-file=dev/terraform.tfvars

-------------------- Resources Created --------------------------------
- VPC with CIDR 10.0.0.0/24

- 2 public subnets 
  10.0.0.0/26     us-east-1a 
  10.0.0.64/26    us-east-1b 

- 2 private subnets 
  10.0.0.128/26  us-east-1a 
  10.0.0.192/26  us-east-1b 

- Gateways 
  Internet Gateway attached to VPC
  NAT Gateway in Public subnets 10.0.0.0/26

- Security Groups 
  Public with 22,80 and 8080 access from 0.0.0.0/0
  Private with 22,80 and 8080 access  from public subnets 

- Load Balancer (Client module)
  Application Load Balancer
     public group access 
     Will be serving public subnets 
  Target Group 
     Port where application is running 
     Health check hyperlink 
  Listener 
     Link Target group 

  *** Repeat Load Balancer for Info module for private subnet  

- AutoScaling 
  Import local key to AWS 
  Create an Instance profile for EC2 
  Create Launch configuration based on Amazon Linux ami
    Attach instanceprofile 
    Add "user_data" to install docker and codedeploy agent 
  Create Auto-scaling group (**** for both client and server)
    Attach launch configuration
    Attach target group create in ALB module 

- CodeDeploy 
  New codedeploy bucket (random postfix)
  Instance profile for codedeploy 

 **** Below tasks are performed for both client and server 
 Create codedeploy application 
  Create codedeploy deployment group 
     group name is the one created in autoscaling module 
  Upload application zipfile to bucket 
  Initiate deploy 
